tenzir/tenzir.cpp

```diff
@@ -306,5 +307,16 @@ auto main(int argc, char** argv) -> int {
     render_error(*root, run_error, std::cerr);
     return EXIT_FAILURE;
   }
+  // Wait for up to 60 seconds for all actors to exit, ignore the rest.
+  if (is_server) {
+    for (int i = 0; i < 600; ++i) {
+      if (sys.registry().running() == 0) {
+        break;
+      }
+      std::this_thread::sleep_for(std::chrono::milliseconds{100});
+    }
+    TENZIR_WARN("Terminating with {} remaining actors",
+                sys.registry().running());
```

Comment:
This won't actually terminate, the destructor of the actor system blocks until we're at zero remaining actors.

Reply:
I didn't test, but according to my understanding of the implementation
```
sys.await_actors_before_shutdown(false);
```
should change that.

Reply:
Isn't it rather dangerous to always set, also for client processes?

Would it perhaps make more sense to patch upstream to be able to set a timeout for the waiting on shutdown?

Reply:
> Would it perhaps make more sense to patch upstream to be able to set a timeout for the waiting on shutdown?

I'd rather formulate that as a feature request and not implement that myself for CAF.